### PR TITLE
Add fake method to Dataset

### PR DESCRIPTION
**Pull request by JouvinLea**
extend the feature fake that was existing for SpectrumDataset to simulate ON counts from Npred. 
Add the method for SpectrumDatasetOnOFF simulated also the OFF and for MapDataset

### CODE REVIEW

File: gammapy/spectrum/dataset.py [hunk 1]
```diff
@@ -343,6 +338,23 @@ def likelihood_per_bin(self):
...
+
+        Parameters
+        ----------
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+                Defines random number generator initialisation.
+                Passed to `~gammapy.utils.random.get_random_state`.
+        """
+        random_state = get_random_state(random_state)
+        data = random_state.poisson(self.npred().data)
+        self.counts.data = data
```
Comment (registerrier) [inline gammapy/spectrum/dataset.py]: Here you need to add the expected background:
```
data = random_state.poisson(self.npred().data + self.background.data)
self.counts.data = data
bkg = random_state.poisson(self.background.data/self.alpha)
self.counts_off.data = bkg
```

File: gammapy/spectrum/dataset.py [hunk 2]
```diff
@@ -343,6 +338,23 @@ def likelihood_per_bin(self):
...
+        Simulate fake data for the background from a Poisson Law of mean value = N background counts in each energy bin.
+        Erase the counts and background data and replace them by the simulated one.
+
+        Parameters
+        ----------
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+                Defines random number generator initialisation.
+                Passed to `~gammapy.utils.random.get_random_state`.
+        """
+        random_state = get_random_state(random_state)
```
Comment (registerrier) [inline gammapy/spectrum/dataset.py]: You should test that `self.background is not None` and possibly raise an exception if it is not there.

**Comment by JouvinLea** [comment]
@registerrier 
Apparently it doesn't like that we define the method fake for the SpetrumDatasetOnOFF with an argument background_model that override the method fake of the parents class SpetrumDataset.

**Comment by adonath** [comment]
@JouvinLea Let's ignore codacy for now...

**Comment by JouvinLea** [comment]
Cool!!

**Comment by JouvinLea** [comment]
@registerrier  ok to merge ?

**Comment by adonath** [comment]
@JouvinLea I just noticed there is a merge conflict in `gammapy/spectrum/tests/test_dataset.py`, can you  resolve it or should I?

**Comment by JouvinLea** [comment]
@adonath it should be fixed now. All my PR are touching the same files so those merge conflict will happen after each merging...
I also notices that the members backscale was changed to acceptance in the SpectrumDatasetOnOFf class but the docs is not updated. do you want me to change it in this PR?

**Comment by adonath** [comment]
Thanks @JouvinLea! No don't change in this PR, I'll clean this up later if needed...

**Comment by adonath** [comment]
Thanks @JouvinLea! I''ll merge this PR now and do a little clean up in a second PR...

**Pull request merged**